Team access changes for editors when editorsCanAdmin is enabled (#45405)
grafana/grafana: docs/sources/http_api/team.md

- All authenticated users are able to view details of teams they are a member of.
- Organization Admins are able to manage all teams and team members.
- If the `editors_can_admin` configuration flag is enabled, Organization Editors are able to create teams and manage teams that they are Admin members of.

>   - Note that editors could find the names of teams that they are not members of by attempting to create a team with the same name.

Reviewer: ```suggestion
  - Editors can find the names of teams they are not members of by creating a team with the same name.
```

Author: Thanks for the review! I've incorporated most of it, but kept in "by **trying** to create a team" rather than "by creating a team". This is because the team creation will fail in this case, and I feel like this nuance is worth calling out.
